### DISCUSSION TITLE
Single model relationship _count

### DISCUSSION THREAD

**Discussion by acip** [laravel/framework#48891] [Ideas]
I am proposing an enhancement to facilitate retrieving the count of related models directly when a single model instance is fetched, say via the `find()` method.  For getting multiple models we already have `withCount`, this idea targets the situations where a single model is retrieved.

This feature would streamline operations, enabling syntax like `auth()->user()->posts_count`. By doing so, developers could easily obtain related model counts without additional queries or method calls.

I propose implementing this feature through a trait, which could introduce a `getRelationshipCount` method. Additionally, a slight alteration in the `Model::_get($key)` method could facilitate this enhancement. Here's a rough outline of how this could be structured:

```
trait HasCountableRelationship
{
  function getRelationshipCount(string $name): ?int
    {
        // logic to get and return the count of the specified relationship
    }
}
```
and in _Model_:
```
class Model {
...
  use HasCountableRelationship;

  public function __get($key)
  {
    ...
    $countableValue = $this->getRelationshipCount($key);
    if (!is_null($countableValue)) {
      return $countableValue;
    }
    ...
  }
...
```
What do you think?

**Comment by gkmk** [comment]
I need to go to sleep but i cant see the value of `auth()->user()->posts_count` vs `auth()->user()->posts()->count()`.
Wouldnt `getRelationshipCount` execute another query?

**Reply by shaedrich** [reply]
Yes, but it would do so only once since it's then a property

**Comment by MadZee-M** [comment]
But the method `->loadCount()` does not help?

```
auth()->user()->loadCount('posts');
auth()->user()->posts_count;
```
